bolt-sidecar/src/api/commitments/firewall/processor.rs

```diff
@@ -0,0 +1,350 @@
+                        this.outgoing_messages.push_front(message);
+                        break;
+                    }
+                    Poll::Ready(Err(e)) => {
+                        error!(?e, "sink error while sending message to websocket");
+                        continue;
+                    }
+                }
+            }
+
+            // 3. Ensure the write sink is flushed so that message are sent to the caller server.
+            //
+            // NOTE: We're not considering "progress" flushing the sink, i.e. `Poll::Ready(())`.
+            // That is because flushing an empty sink would lead to run this loop indefinitely
+            if let Poll::Ready(Err(e)) = this.write_sink.poll_flush_unpin(cx) {
+                error!(?e, "failed to flush websocket write sink");
+                continue;
+            }
+
+            // 4. Handle shutdown signals before accepting any new work
+            match this.shutdown_rx.try_recv() {
+                Ok(_) => {
+                    info!("received shutdown signal. closing websocket connection...");
+                    return Poll::Ready(InterruptReason::Shutdown);
+                }
+                Err(TryRecvError::Closed) => {
+                    error!("shutdown channel closed");
+                }
+                _ => {}
+            }
+
+            // Incoming work tasks
+
+            // 5. Handle incoming websocket messages from the read stream.
+            while let Poll::Ready(maybe_message) = this.read_stream.poll_next_unpin(cx) {
+                progress = true;
+
+                let Some(res_message) = maybe_message else {
+                    warn!(?rpc_url, "websocket read streaam terminated");
+                    return Poll::Ready(InterruptReason::ReadStreamTerminated);
+                };
+
+                match res_message {
+                    Ok(Message::Text(text)) => {
+                        trace!(?rpc_url, text, "received text message from websocket connection");
+                        // Create the channel to send and receive the commitment response
+                        let (tx, rx) = oneshot::channel();
+
+                        let request = serde_json::from_str::<JsonPayloadUuid>(&text)
+                            .map_err(|e| e.to_string());
+
+                        let response = match request {
+                            Err(e) => Err(e),
+                            Ok(request) => {
+                                let id = request.id;
+
+                                match request.method.as_str() {
+                                    GET_VERSION_METHOD => Ok(JsonResponse {
+                                        id: Some(Value::String(id.to_string())),
+                                        result: Value::String(BOLT_SIDECAR_VERSION.clone()),
+                                        ..Default::default()
+                                    }),
```

Comment:
from [spec](https://www.jsonrpc.org/specification)

> jsonrpc
    A String specifying the version of the JSON-RPC protocol. MUST be exactly "2.0". 

```suggestion
                                    GET_VERSION_METHOD => Ok(JsonResponse {
                                        jsonrpc: "2.0".to_string(),
                                        id: Some(Value::String(id.to_string())),
                                        result: Value::String(BOLT_SIDECAR_VERSION.clone()),
                                        ..Default::default()
                                    }),
```

Reply:
The `..Default::default()` already takes care of that